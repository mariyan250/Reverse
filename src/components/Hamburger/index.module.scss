@import '../../index.scss';

.hamburger {
  width: to-rem(35px);
  height: to-rem(30px);

  span {
    height: to-rem(4px);
    transition: $timing-slow;
  }

  span:nth-child(1) {
    width: 100%;
  }

  span:nth-child(2) {
    width: to-rem(22px);
  }

  span:nth-child(3) {
    width: to-rem(28px);
  }

  &:hover {
    cursor: pointer;
    outline: none;
  }

  &.toggled {
    span:nth-child(1) {
      transform: rotate(-45deg) translate(-25%, 235%);
    }

    span:nth-child(2) {
      opacity: 0;
    }

    span:nth-child(3) {
      transform: rotate(45deg) translate(-25%, -235%);
      width: 100%;
    }
  }
}
