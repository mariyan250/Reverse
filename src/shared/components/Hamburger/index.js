// React and Style
import React from 'react';
import './index.scss';

// krastavici?
function Hamburger(props) {
  return (
    <section {...props}>
      <article className="hamburger d-flex flex-column justify-content-between">
        <span></span>
        <span></span>
        <span></span>
      </article>
    </section>
  );
}

export default Hamburger;
