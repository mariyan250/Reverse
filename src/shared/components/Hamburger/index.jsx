// React and Style
import React from 'react';
import styles from './index.module.scss';

function Hamburger({ isOpen, onClick }) {
  return (
    <section
      className={`${styles.hamburger} ${
        isOpen && styles.toggled
      } d-flex flex-column justify-content-between`}
      onClick={onClick}
      data-testid="hamburger"
    >
      <span className={`app-bg-white rounded-pill`}></span>
      <span className={`app-bg-white align-self-end rounded-pill`}></span>
      <span className={`app-bg-white align-self-end rounded-pill`}></span>
    </section>
  );
}

export default Hamburger;
