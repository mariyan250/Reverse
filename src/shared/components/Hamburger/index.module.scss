@import '../../../index.scss';

.hamburger {
  width: to-rem(35px);
  height: to-rem(28px);

  span {
    height: to-rem(4px);
    transition: $slow;
  }

  span:nth-child(2) {
    width: to-rem(22px);
  }

  span:nth-child(3) {
    width: to-rem(28px);
  }

  &:hover {
    cursor: pointer;
  }

  &.toggled {
    span:nth-child(1) {
      transform: rotate(-45deg) translate(-20%, 235%);
    }

    span:nth-child(2) {
      opacity: 0;
    }

    span:nth-child(3) {
      transform: rotate(45deg) translate(-20%, -235%);
      width: to-rem(35px);
    }
  }
}
