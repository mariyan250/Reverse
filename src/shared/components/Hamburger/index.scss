@import '../../../index.scss';

.hamburger {
  width: 2.5rem;
  height: 2rem;

  span {
    width: 100%;
    height: 0.3rem;
    background: $background-white;
    transition: 0.5s;
  }

  &:hover {
    cursor: pointer;
  }
}

.toggled {
  .hamburger {
    span:nth-child(1) {
      transform: rotate(-45deg) translate(-20%, 235%);
    }

    span:nth-child(2) {
      opacity: 0;
    }

    span:nth-child(3) {
      transform: rotate(45deg) translate(-20%, -235%);
    }
  }
}
