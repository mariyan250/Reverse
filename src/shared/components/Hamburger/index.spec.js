import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render, screen } from 'tests/test-utils';
import Hamburger from './index';

describe('Hamburger', () => {
  it('should render the component', () => {
    const { container } = render(<Hamburger />);
    expect(container).toMatchSnapshot();
  });

  it('should change style when open', () => {
    const { container } = render(<Hamburger isOpen={true} />);
    const expected = {
      '0': 'hamburger',
      '1': 'toggled',
      '2': 'd-flex',
      '3': 'flex-column',
      '4': 'justify-content-between',
    };
    expect(screen.getByTestId('hamburger').classList).toHaveProperty('1', 'toggled');
  });
});
