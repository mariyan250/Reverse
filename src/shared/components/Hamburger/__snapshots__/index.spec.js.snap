// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger should render the component 1`] = `
<div>
  <section
    class="hamburger undefined d-flex flex-column justify-content-between"
    data-testid="hamburger"
  >
    <span
      class="app-bg-white rounded-pill"
    />
    <span
      class="app-bg-white align-self-end rounded-pill"
    />
    <span
      class="app-bg-white align-self-end rounded-pill"
    />
  </section>
</div>
`;
